Feature/codecheck metadata form

subhanLabs opened the pull request:
Add CODECHECK metadata form to review workflow

- Create dedicated CODECHECK tab in workflow sidebar under Review stage
- Add form for capturing CODECHECK configuration version, publication method
- Implement manifest table for output files and descriptions
- Add paper metadata fields (title, authors with ORCIDs, reference)
- Add source material URL input
- Add CODECHECKers list with ORCID support
- Add CODECHECK certificate URL field with upload functionality
- Add summary textarea for CODECHECK completion notes
- Add repositories list for code/data links
- Add completion timestamp and identifier fields
- Integrate form with Vue.js components for dynamic interactions

Closes #9 

Closes #30
---

## Related User Stories

- C-MM1 (related): As a codechecker, I want the plugin to provide a form that generates, validates, and stores the codecheck.yml...
- P-IM1 (related): As a journal manager, I want to enter, edit, and view CODECHECK metadata entirely within the OJS editorial workflow...

_User story references from the [CHECK-PUB User Stories document](https://codecheck.org.uk/pub/)._

subhanLabs committed:
feat: Implement CODECHECK workflow metadata form

Add comprehensive CODECHECK form in review stage with manifest management, certificate tracking, and repository links.

subhanLabs committed:
Merge branch 'main' into feature/codecheck-metadata-form

nuest commented on resources/js/Components/CodecheckReviewDisplay.vue:
```diff
@@ -0,0 +1,206 @@
+  return repos;
+});
+
+const statusClass = computed(() => {
```
You are doubling the cases from `getStatus(..)` here; I think it is a good idea though to seperate the internal status name from the CSS class, so instead, let `statusClass` rely on the response of `getStatus()` and translate that to the right class, maybe with a `switch` statement.

nuest commented on resources/js/main.js:
```diff
@@ -0,0 +1,267 @@
+    createApp(CodecheckManifestFiles, {
+      name: 'manifestFiles',
+      label: 'Expected Output Files',
+      description: 'List the main figures, tables, and results',
```
Expand description:

"List the main figures, tables, and results that the codechecker should consider for their evaluation."

Also, this description needs to be translated! Find out how you can load strings from `locale.po` into Vue.js components...

nuest commented on resources/js/main.js:
```diff
@@ -0,0 +1,267 @@
+    createApp(CodecheckRepositoryList, {
+      name: 'dataRepository',
+      label: 'Data Repository URLs',
+      description: 'Link(s) to your data repository',
```
See above.

subhanLabs committed:
feat: Refactor CODECHECK metadata architecture and implement dynamic handler

BREAKING CHANGES:
- Migrate metadata storage from submission-level to publication-level
- Restructure database schema (codecheck_metadata table)
- Move fields (codeRepository, dataRepository, etc.) to publications table

Core Changes:
- Add CodecheckMetadataHandler for dynamic API endpoints (GET/POST/YAML)
- Implement LoadHandler hook for custom routing (page=codecheck)
- Refactor metadata form to load data via AJAX instead of props
- Add proper error handling and loading states

Database:
- Drop old codecheck_metadata table structure
- Create new schema with check_time, report_url, and repository fields
- Update migration to handle existing installations safely

UI/UX Improvements:
- Display submission metadata as read-only in form
- Add real-time data loading with error recovery
- Improve form validation (manifest, codecheckers, certificate)
- Add YAML preview modal with download functionality
- Enhance status badges (complete/in-progress/pending)

API:
- Add metadata retrieval endpoint (GET /codecheck/metadata)
- Add metadata save endpoint (POST /codecheck/metadata)
- Add YAML generation endpoint (GET /codecheck/yaml)
- Implement proper CSRF token handling

Components:
- Update CodecheckReviewDisplay to show database values
- Refactor CodecheckMetadataForm for async data loading
- Add proper localization throughout (using useLocalize)
- Improve repository and manifest file management

Bug Fixes:
- Fix workflow menu integration (stageId: 999)
- Remove localStorage usage (not supported in OJS)
- Fix Vue 3 compatibility issues
- Correct ORCID badge display
- Fix datetime-local input formatting

Styling:
- Minimize CSS to essential plugin styles
- Remove duplicate declarations
- Improve form layout and spacing
- Add proper button styling (btn-add, btn-remove)

Developer Experience:
- Add comprehensive error logging
- Improve debugging output
- Document frontend build process in README
- Add validation helpers

Files Changed: 17 files
- Core: CodecheckPlugin.php, CodecheckMetadataHandler.php
- Migration: CodecheckSchemaMigration.php
- Frontend: CodecheckMetadataForm.vue, CodecheckReviewDisplay.vue
- Locale: locale.po (144 new translation keys)
- Build: public/build/* (compiled assets)

subhanLabs committed:
Refactor CODECHECK metadata form UI and improve user experience

- Update form layout and styling for better visual hierarchy
- Improve localization and text clarity
- Enhance form component organization
- Update build assets and locale registry
- Maintain all existing functionality while improving usability

subhanLabs committed:
Add file upload/download to CODECHECK manifest

- Implement server-side upload/download handlers
- Make manifest filenames clickable for download
- Store files in dedicated codecheck directory
- Add file path tracking in manifest metadata

nuest commented on classes/Workflow/CodecheckMetadataHandler.php:
```diff
@@ -0,0 +1,337 @@
+                'certificate' => $metadata->certificate,
+                'checkTime' => $metadata->check_time,
+                'summary' => $metadata->summary,
+                'reportUrl' => $metadata->report_url,
```
Please match field names closely here to the config file spec, where the field is called "report" only.

Subhanaliweb replied:
Yes they are exactly the same as config, these are just variable are little refactor like camel case, but in yml it was generating the same config labels, but I also match the variable now.

<img width="772" height="688" alt="image" src="https://github.com/user-attachments/assets/778d6673-284e-4420-a726-76baa1315272" />

nuest reviewed (changes requested):
Please add some visual cue the non editable fields, maybe grey them out? The user should not be confused between something being "broken" and something being not allowed.

Please comment out (or make unselectable) the "Check published in a section of the journal", because we are not going to support that in the foreseeable future (#21).

subhanLabs committed:
Revert "Add file upload/download to CODECHECK manifest"

This reverts commit 00ad56f81891f120806dff762f8391690bf7ce66.

subhanLabs committed:
Add read-only indicators to Paper Metadata fields

- Add description banner explaining fields are auto-populated
- Add tooltips to Title and Authors fields indicating read-only status
- Add locale keys for read-only messaging

nuest commented:
@Subhanaliweb I converted the PR to a Draft - please get in touch when you have handled all feedback and requests. Thanks!

Subhanaliweb commented:
> Please add some visual cue the non editable fields, maybe grey them out? The user should not be confused between something being "broken" and something being not allowed.
> 
> Please comment out (or make unselectable) the "Check published in a section of the journal", because we are not going to support that in the foreseeable future (#21).

<img width="1025" height="109" alt="image" src="https://github.com/user-attachments/assets/1c44625a-5dab-4c29-ba4c-cd17cfd47a9c" />

Shall I comment down only the second option? and keep the first one?

subhanLabs committed:
Add source and additional_content fields, use YAML library for generation

- Add source field for material provenance description
- Add additional_content field for custom YAML metadata
- Replace manual YAML construction with Symfony YAML library
- Separate author name retrieval into given/family names
- Update form, preview, and database schema accordingly

Subhanaliweb commented on resources/js/main.js:
```diff
@@ -0,0 +1,267 @@
+    createApp(CodecheckRepositoryList, {
+      name: 'codeRepository',
+      label: 'Code Repository URLs',
+      description: 'Link(s) to your code repository',
```
```
    createApp(CodecheckRepositoryList, {
      name: 'codeRepository',
      label: t('plugins.generic.codecheck.codeRepository'),
      description: t('plugins.generic.codecheck.codeRepository.description'),
      value: textarea.value,
    }).mount(vueDiv);
```

Updated.

Subhanaliweb commented on resources/js/main.js:
```diff
@@ -0,0 +1,267 @@
+  props: ['file'],
+  computed: {
+    statusText() {
+      // TODO: Implement actual status checking
```
I have removed such comments.

Subhanaliweb commented on classes/Workflow/CodecheckMetadataHandler.php:
```diff
@@ -0,0 +1,337 @@
+        $authors = [];
+        foreach ($publication->getData('authors') as $author) {
+            $authors[] = [
+                'name' => $author->getFullName(),
```
```
        foreach ($publication->getData('authors') as $author) {
            $locale = $author->getDefaultLocale();
            $givenName = $author->getGivenName($locale) ?? '';
            $familyName = $author->getFamilyName($locale) ?? '';
            $fullName = trim($givenName . ' ' . $familyName);
            
            $authorData = ['name' => $fullName];
            if ($author->getOrcid()) {
                $authorData['ORCID'] = $author->getOrcid();
            }
            $authors[] = $authorData;
        }
```

Done.

subhanLabs committed:
Replace manual YAML generation with js-yaml library

- Use js-yaml in Vue component for safe YAML preview generation
- Add document separator (---) to PHP YAML output
- Ensure consistent YAML formatting between frontend and backend

Subhanaliweb commented:
@nuest I have made all requested changes please feel free to review and merge.

nuest reviewed (approved):
Happy to merge this - @Subhanaliweb please resolve the merge conflicts and ping me again.

dxL1nus commented:
> Happy to merge this - @Subhanaliweb please resolve the merge conflicts and ping me again.

@nuest my branch is up to date with this codecheck-metadata branch of Subhan. So as soon as this gets merges, my PR can be merged as well

subhanLabs committed:
Merge branch 'main' into feature/codecheck-metadata-form
